pkg/query/v2/logical/plan_traceid.go

```diff
@@ -0,0 +1,193 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logical
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	commonv2 "github.com/apache/skywalking-banyandb/api/proto/banyandb/common/v2"
+	streamv2 "github.com/apache/skywalking-banyandb/api/proto/banyandb/stream/v2"
+	"github.com/apache/skywalking-banyandb/banyand/tsdb"
+	"github.com/apache/skywalking-banyandb/pkg/index"
+	"github.com/apache/skywalking-banyandb/pkg/query/v2/executor"
+)
+
+var _ UnresolvedPlan = (*unresolvedTraceIDFetch)(nil)
+var _ Plan = (*traceIDFetch)(nil)
+
+type unresolvedTraceIDFetch struct {
+	metadata         *commonv2.Metadata
+	traceID          string
+	projectionFields [][]*Tag
+}
+
+func (t *unresolvedTraceIDFetch) Analyze(s Schema) (Plan, error) {
+	defined, idxRule := s.IndexDefined(NewTag("", s.TraceIDFieldName()))
```

Comment:
We don't have to parse trace_id as I mentioned above, parse global index instead.

Reply:
GlobalIndexScan has been added.